versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -1060,15 +1097,15 @@ protected Hash copyCommits(
               newKeyLists);
       keyListDistance = newEntry.getKeyListDistance();
 
-      if (!newEntry.getHash().equals(sourceCommit.getHash())) {
+      if (!newEntry.getHash().equals(sourceCommitValueHash)) {
```

Comment:
Why this change and the ones below?

Reply:
Because of https://github.com/projectnessie/nessie/pull/2146#discussion_r719170422

Reply:
The newEntry is created with reset version of commit metadata. The hash won't match in intended cases with the original source metadata hash. Hence, recomputing the source metadata hash and use the same.